### PR TITLE
Fix markdown link check 

### LINKED ISSUE #1862: Broken links: configure link checker

### PR DESCRIPTION
**Pull request by Dxuian**
regarding #2480

fixes #1862 

refs #344

### CODE REVIEW

**Comment by Dxuian** [comment]
@johnbampton  @BaseMax  okay so there's nothing wrong with the link checker the problem seems to be in the fact that the links it tries to check are actually dead or behind a 301, 302, 307, and 308 redirect the solution to this should be removal of dead links and accommodation of redirects

**Comment by Dxuian** [comment]
403 codes are also present in some links that work

**Comment by Dxuian** [comment]
i've looked up the GitHub action and looks like there's a alternative to it  so i will be trying that one

**Comment by Dxuian** [comment]
> i've looked up the GitHub action and looks like there's a alternative to it so i will be trying that one

looks like the alternative isnt fully built so its better to stick to this

**Comment by Dxuian** [comment]
@johnbampton  @BaseMax  can you add hacktoberfest label to this issue please 🙏🙏🙏🙏🙏🙏

**Comment by Dxuian** [comment]
@johnbampton  please review

File: mlc_config.json [hunk 1]
```diff
@@ -7,5 +7,30 @@
       }
     }
   ],
-  "aliveStatusCodes": [200, 429]
+  "timeout": "20s",
+  "retryOn429": true,
+  "retryCount": 5,
+  "fallbackRetryDelay": "30s",
+  "aliveStatusCodes": [200, 206, 429, 300, 301, 302, 303, 304, 305, 306, 307, 308, 403, 999],
```
Comment (jbampton) [inline mlc_config.json:14]:
```suggestion
  "aliveStatusCodes": [200, 206, 429, 300, 301, 302, 303, 304, 305, 306, 307, 308, 403, 999],
```

Please order these

Comment (Dxuian) [reply mlc_config.json]: @johnbampton  done please review

**Comment by Dxuian** [comment]
@jbampton  done please review

File: charts/lang/README.de.md [hunk 2]
```diff
@@ -24,7 +24,6 @@ Die plotly.js Tortendiagramme sind die mit Abstand interaktivsten von allen.
 
 Sogar die README.md-Datei wird aus der config.xml-Konfigurationsdatei erstellt.
 
-Eine Demo finden Sie hier: <http://thebeast.me/charts/>
```
Comment (jbampton) [inline charts/lang/README.de.md:27]:
```suggestion
```

File: doc/source/dita-to-pdf-with-eclipse.md [hunk 3]
```diff
@@ -148,8 +148,6 @@ Mastering Markdown - <https://guides.github.com/features/mastering-markdown/>
 
 GitHub Desktop - <https://desktop.github.com/>
```
Comment (jbampton) [inline doc/source/dita-to-pdf-with-eclipse.md:150]:
```suggestion
```

File: doc/source/goa.md [hunk 4]
```diff
@@ -45,7 +45,6 @@ The Every Noise at Once platform provides an opportunity for people to get to
 know better the names of the creators who have ever been in the genre and thus
 the peculiarities of it.
 
-<https://everynoise.com/everynoise1d.cgi?root=goa%20psytrance&scope=all>
```
Comment (jbampton) [inline doc/source/goa.md:48]:
```suggestion
```

File: doc/source/jungle-drum-bass.md [hunk 5]
```diff
@@ -18,7 +18,6 @@ You can find more details about its formation and development on Ishkur's Guide
 
 The Every Noise at Once platform provides an opportunity for people to get to know better the names of the creators who have ever been in the genre and thus the peculiarities of it.
```
Comment (jbampton) [inline doc/source/jungle-drum-bass.md:20]:
```suggestion
```

File: doc/source/sql-file.md [hunk 6]
```diff
@@ -52,11 +52,8 @@ front-end for working on a website.
 
 For example, go to:
 
-<http://www.sitename.com/controlpanel>
+<https://id.cpanel.net/get/login?url=aHR0cHM6Ly9tYW5hZ2UyLmNwYW5lbC5uZXQvbG9naW4uY2dp>
```
Comment (jbampton) [inline doc/source/sql-file.md:56]:
```suggestion
```

File: doc/source/xml.md [hunk 7]
```diff
@@ -43,8 +43,6 @@ processor
 [lxml](https://lxml.de/) - the most feature-rich and easy-to-use library for
 processing XML and HTML in the Python language.
 
-[EXSLT](http://exslt.org/) - is a community initiative to provide extensions to
-XSLT.
```
Comment (jbampton) [inline doc/source/xml.md:46]:
```suggestion
```

File: mlc_config.json [hunk 8]
```diff
@@ -7,5 +7,30 @@
...
+      "pattern": "^https://lawandorder.fandom.com/"
+    },
+    {
+      "pattern": "^https://upload.wikimedia.org/"
+    },
+    {
+      "pattern": "^https://www.adobe.com/au/acrobat/about-adobe-pdf.html"
+    }
+  ]
+}
```
Comment (jbampton) [inline mlc_config.json:36]:
```suggestion
}

```

File: doc/source/trip-hop.md [hunk 9]
```diff
@@ -22,7 +22,6 @@ You can find more details about its formation and development on Ishkur's Guide
 
 The Every Noise at Once platform provides an opportunity for people to get to know better the names of the creators who have ever been in the genre and thus the peculiarities of it.
 
-<https://everynoise.com/everynoise1d.cgi?root=trip%20hop&scope=all>
```
Comment (jbampton) [inline doc/source/trip-hop.md:25]:
```suggestion
```

File: doc/source/trance.md [hunk 10]
```diff
@@ -27,7 +27,6 @@ music right away.
 
 The Every Noise at Once platform provides an opportunity for people to get to know better the names of the creators who have ever been in the genre and thus the peculiarities of it.
 
-<https://everynoise.com/everynoise1d.cgi?root=trance&scope=all>
```
Comment (jbampton) [inline doc/source/trance.md:30]:
```suggestion
```

**Comment by Dxuian** [comment]
@johnbampton  i made a change please review and merge,  the test works fine now

File: mlc_config.json [hunk 11]
```diff
@@ -7,5 +7,33 @@
...
+      "pattern": "^https://upload.wikimedia.org/"
+    },
+    {
+      "pattern": "^https://www.adobe.com/au/acrobat/about-adobe-pdf.html"
+    },
+    {
+      "pattern":"^https://www.emacswiki.org/"
+    }
+  ]
+}
```
Comment (jbampton) [inline mlc_config.json:39]:
```suggestion
}

```

**Comment by Dxuian** [comment]
@jbampton  the test works fine now can we merge and close this #1862  and #2480

File: README.md [hunk 12]
```diff
@@ -11,7 +10,6 @@
 
 [![Slurp is GPL-3.0](https://img.shields.io/github/license/slurpcode/slurp.svg)](https://github.com/slurpcode/slurp/blob/main/LICENSE)
 [![This project is using Percy.io for visual regression testing](https://percy.io/static/images/percy-badge.svg)](https://percy.io)
-[![Codacy Badge](https://app.codacy.com/project/badge/Grade/4199d4805b2a473284988680cffe85db)](https://www.codacy.com/gh/slurpcode/slurp/dashboard)
```
Comment (jbampton) [inline README.md:14]: https://app.codacy.com/gh/slurpcode/slurp/dashboard

Need to get the correct working badge

Codacy runs with GitHub actions on PR

Screenshot from 2024-10-14 01-43-02

Comment (Dxuian) [reply README.md:14]: ive added it back and it seems to be working pls review

**Comment by Dxuian** [comment]
@jbampton any other changes?

File: README.md [hunk 13]
```diff
@@ -1,7 +1,6 @@
 # slurp
 
 [![Fuchsia Programming](https://img.shields.io/website?label=github%20pages&url=https%3A%2F%2Fslurpcode.github.io%2Fslurp)](https://slurpcode.github.io/slurp)
-[![Logorama](https://img.shields.io/website?label=heroku&url=https%3A%2F%2Flogorama.herokuapp.com%2F)](https://logorama.herokuapp.com/)
```
Comment (jbampton) [inline README.md:4]: This link should not be removed but just commented out

File: doc/source/sql-file.md [hunk 14]
```diff
@@ -52,11 +52,7 @@ front-end for working on a website.
 
 For example, go to:
 
-<http://www.sitename.com/controlpanel>
-
-or here:
-
-<http://www.sitename.com:2082>
+<https://id.cpanel.net/get/login?url=aHR0cHM6Ly9tYW5hZ2UyLmNwYW5lbC5uZXQvbG9naW4uY2dp>
```
Comment (jbampton) [inline doc/source/sql-file.md:55]:
```suggestion
<https://id.cpanel.net>
```

File: README.md (hunk 13 above)
Comment (Dxuian) [reply README.md:4]: done @jbampton

**Comment by Dxuian** [comment]
@jbampton any changes?

**Comment by Dxuian** [comment]
@johnbampton  @BaseMax  it passes all tests can you merge and close issue

**Comment by jbampton** [comment]
Hey @Dxuian can you please comment on this issue to say you want it ?

https://github.com/slurpcode/slurp/issues/1862

File: charts/assets/js/d3/CHANGES.md
Comment (jbampton) [inline charts/assets/js/d3/CHANGES.md:1]: The `charts/assets` directory should be excluded from checking.

This file should not be changed.

You can add an "exclude" to pre-commit

Comment (Dxuian) [reply charts/assets/js/d3/CHANGES.md:1]: done

**Comment by Dxuian** [comment]
@johnbampton  it passes the link test now pls review

File: charts/assets/js/d3/CHANGES.md
Comment (jbampton) [inline charts/assets/js/d3/CHANGES.md:1]: This file should not be changed.

All charts/assets files should be excluded.

**Comment by Dxuian** [comment]
the test is failing again lemme see

**Comment by Dxuian** [comment]
since dir exclusion isnt possible i have found a alternative to this action its basically the newer version of the old one so im going to experiment with the new one

**Comment by Dxuian** [comment]
> since dir exclusion isnt possible i have found a alternative to this action its basically the newer version of the old one so im going to experiment with the new one

the tool is still incomplete reverting to old one

**Comment by Dxuian** [comment]
@johnbampton  ive managed to fix pre-commit errs also

**Comment by Dxuian** [comment]
@johnbampton  any changes ?

**Pull request merged**